app/helpers/automated_tests_helper.rb

```diff
@@ -85,61 +118,122 @@ def create_ant_test_files(assignment)
   def process_test_form(assignment, params)
 
     # Hash for storing new and updated test files
-    updated_files = {}
-
-    # Retrieve all test file entries
-    testfiles = params[:test_files_attributes]
-
-    # First check for duplicate filenames:
-    filename_array = []
-    testfiles.values.each do |tfile|
-      if tfile['filename'].respond_to?(:original_filename)
-        fname = tfile['filename'].original_filename
-        # If this is a duplicate filename, raise error and return
-        if filename_array.include?(fname)
-          raise I18n.t('automated_tests.duplicate_filename') + fname
-        else
-          filename_array << fname
+    updated_script_files = {}
+    updated_support_files = {}
+
+    # Array for checking duplicate file names
+    file_name_array = []
+
+    #add existing scripts names
+    params.each {|key, value| if(key[/test_script_\d+/] != nil) then file_name_array << value end}
+
+    # Retrieve all test scripts
+
+
+    testscripts = params[:test_scripts_attributes]
+
+
+
+    # First check for duplicate script names in test scripts
+    if !testscripts.nil?
+      testscripts.values.each do |tfile|
+        if tfile['script_name'].respond_to?(:original_filename)
+          fname = tfile['script_name'].original_filename
+          # If this is a duplicate script name, raise error and return
+          if !file_name_array.include?(fname)
+            file_name_array << fname
+          else
+            raise I18n.t("automated_tests.duplicate_filename") + fname
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.